core/src/main/scala/chisel3/IO.scala

```diff
@@ -32,8 +33,8 @@ object IO {
     module match {
       case _: Class => {
         data match {
-          case _: Data => throwException(s"Class ports must be Property type, but found $data")
-          case _ => ()
+          case _: Property[_] => ()
+          case _ => throwException(s"Class ports must be Property type, but found $data")
```

Comment:
Since we are here, can we throw an non-blocking error?

Reply:
Non-blocking meaning recoverable so that we can aggregate more errors? Sure.